### PR TITLE
Spring 01

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -64,5 +99,4 @@
             </plugins>
         </pluginManagement>
     </build>
-
 </project>
```
Comment (DmGrt) [inline pom.xml]: You need an empty line here

**Review by DmGrt — COMMENTED** [review]
Good job! Just add one empty line.

File: pom.xml (hunk 1 above)
Comment (vremere-a) [reply pom.xml]: done

File: src/main/java/com/spring/intro/config/AppConfig.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        dataSource.setUsername("root");
+        dataSource.setPassword("Vremere_a");
+        return dataSource;
+    }
+
+    @Bean
+    public LocalSessionFactoryBean getSessionFactory() {
+        LocalSessionFactoryBean factoryBean = new LocalSessionFactoryBean();
+        factoryBean.setDataSource(getDataSource());
+        Properties props = new Properties();
```
Comment (Sofasmile) [inline src/main/java/com/spring/intro/config/AppConfig.java]:
```suggestion
        Properties properties = new Properties();
```

File: src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@Log4j
+public class UserDaoImpl implements UserDao {
+
+    @Autowired
+    private SessionFactory sessionFactory;
```
Comment (Sofasmile) [inline src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java]: What is better: setter, constructor, or field injection?

File: src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+            log.info("Entity " + user + "successful added from the DB");
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add user entity with id"
+                    + user.getId(), e);
```
Comment (Sofasmile) [inline src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java]: do we know user id here?

File: src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<User> listUsers() {
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder cb = session.getCriteriaBuilder();
```
Comment (Sofasmile) [inline src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java]: cb -?

File: src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+            CriteriaQuery<User> cq = cb.createQuery(User.class);
```
Comment (Sofasmile) [inline src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java]: cq -?

File: src/main/java/com/spring/intro/service/impl/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import com.spring.intro.service.UserService;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Autowired
+    private UserDao userDao;
```
Comment (Sofasmile) [inline src/main/java/com/spring/intro/service/impl/UserServiceImpl.java]: the same)

File: src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java (hunk 3 above)
Comment (vremere-a) [reply src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java]: constructor

File: src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java (hunk 4 above)
Comment (vremere-a) [reply src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java]: no

File: src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java (hunk 5 above)
Comment (vremere-a) [reply src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java]: donr

File: src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java (hunk 6 above)
Comment (vremere-a) [reply src/main/java/com/spring/intro/dao/impl/UserDaoImpl.java]: done

File: src/main/java/com/spring/intro/service/impl/UserServiceImpl.java (hunk 7 above)
Comment (vremere-a) [reply src/main/java/com/spring/intro/service/impl/UserServiceImpl.java]: ok

File: src/main/java/com/spring/intro/config/AppConfig.java (hunk 2 above)
Comment (vremere-a) [reply src/main/java/com/spring/intro/config/AppConfig.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**